src/cpp/session/SessionMain.cpp

```diff
@@ -1642,6 +1643,12 @@ Error runPreflightScript()
    // always return success
    return Success();
 }
+
+void exitEarly(int status)
+{
+   FileLock::cleanUp();
+   ::exit(status);
```

Comment:
Does this imply that we're going to be broken if someone does a naked `::exit()`? If so it might be worth adding something to `FileLock.hpp` that shadows `::exit` with something that does the right thing (or just barfs and asks you to call `exitEarly`).